### PR TITLE
Add PR114 RunPod H100 SXM non-record submission

### PR DESCRIPTION
**Pull request by greqone**
This PR now submits a **non-record** RunPod-verified package for the PR114 long-context selective-precision recipe.

Why this changed:
- the earlier qscale draft was not clean enough to take out of draft status
- after that, the public leaderboard moved again; as of March 21, 2026 the live README shows `1.1428` as the top score
- the new package is therefore intentionally framed as a non-record submission

What is included:
- one new folder under `records/track_non_record_16mb/2026-03-21_PR114_RunPod_H100SXM_TripleRerun`
- self-contained `train_gpt.py`
- `train.log` for the strongest same-provider RunPod rerun
- two additional same-provider rerun logs
- `submission.json` and `significance_runs.tsv`

Best packaged run:
- `val_bpb = 1.15536852`
- `val_loss = 1.95079135`
- `bytes_total = 15963195`
- `bytes_model_int8_zlib = 15909623`
- `bytes_code = 53572`

Same-provider reruns on RunPod `8x H100 SXM`:
- `seed=1337` -> `1.15862019`
- `seed=1338` -> `1.15536852`
- `seed=1339` -> `1.15563461`

These three reruns are all under the decimal `16,000,000` byte cap and give one-sided `p = 0.00306` against the older `1.1698` threshold, but this PR is **not** claiming a live SOTA record because the leaderboard moved past it.

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
> **[RETRACTED 2026-04-11]** — This IMPORT_FAIL was a false positive. Root cause: classifier ast.parse() didn't handle UTF-8 BOM; smoke test actually PASSED. Your code is not broken. See correction below: https://github.com/openai/parameter-golf/pull/252#issuecomment-4230238683

---

### Community Review — Add PR114 RunPod H100 SXM non-record submission

**Compliance:** NEEDS AUTHOR ACTION — `train_gpt.py` fails to import on CT2038 (Python 3.10 / torch 2.10.0+cpu)

**What I found:** The CPU smoke test on CT2038 (proteus-engine, 128 GB RAM, Triton 3.6.0, flash_attn stub, cutlass_evt_fusion stub) failed at the import step with:

```
syntax error at line 1: invalid non-printable character U+FEFF
```

A few of the common patterns I've seen for this class of error in the 2026-04-11 sweep:

- **PEP 701 f-string nesting** — e.g. `log(f"  {cat}: {", ".join(...)}")` is valid Python 3.12+ but invalid Python 3.10 because the inner `", "` re-enters the outer double-quote context. One-character fix: `', '` instead of `", "`. See PR #1541 / #1523 for reference.
- **Missing flash_attn variants** — e.g. `from flash_attn_interface import flash_attn_varlen_func` when the wrapper script only stubs `flash_attn_func`. Not a PR defect on H100s, but the eval image / CPU preflight path needs a guarded import.
- **Local compiled extension** — e.g. `import cutlass_evt_fusion` from a `records/*/cutlass_evt_fusion/` subfolder that isn't on the import path at smoke time. Usually an import-order issue inside the script.
- **Actual syntax error** — typo, missing bracket, etc.

**Recommendation:** Could you run `python3 -c "import py_compile; py_compile.compile('train_gpt.py')"` on your records-folder `train_gpt.py` under **Python 3.10** specifically? The eval image is Python 3.10 per Issue #17 / the README, so any parse error on 3.10 blocks the submission at import time before any of the scored-eval logic runs.

Once the parse/import issue is fixed, I'll re-run the compliance audit through the normal pipeline. No other flags identified yet because the audit halts at the import step.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): **IMPORT_FAIL** — syntax error at line 1: invalid non-printable character U+FEFF. Classification via `classify_prs.py` AST-based classifier; full compliance audit deferred until the import issue is resolved. Auto-drafted from a template and spot-checked before posting.*

**Comment by MatoTeziTanka** [comment]
### Retraction — this IMPORT_FAIL was a UTF-8 BOM handling bug in my classifier

Sorry @greqone, this one's on me. Your CPU smoke test on CT2038 actually **passed** — the IMPORT_FAIL I reported above came from a separate classifier step, and it was a bug in the classifier, not in your code.

**What happened:**

My classifier does an `ast.parse()` walk over your file to check for n-gram family bugs, SLOT masks, Pre-Quant TTT patterns, etc. It opened `records/track_non_record_16mb/2026-03-21_PR114_RunPod_H100SXM_TripleRerun/train_gpt.py` in plain UTF-8 mode, and your file starts with a UTF-8 byte-order mark (`U+FEFF`, bytes `EF BB BF`). Python's own parser handles BOMs fine via `encoding='utf-8-sig'`, but my classifier used plain UTF-8, so `ast.parse()` threw `invalid non-printable character U+FEFF at line 1`.

The smoke runner's `importlib` path (which uses Python's real loader) imported your file without issue and reported:

```
SMOKE_TEST_PASS — import OK, HAS_HYPERPARAMETERS=True, HAS_GPT=True
```

**Your PR is not broken.** Python accepts BOMs; my classifier's ast walk was buggy. I'm retracting the IMPORT_FAIL classification. I'll re-queue the compliance audit now that the BOM-handling bug is identified and post findings separately.

Again — sorry for the noise.

**Comment by MatoTeziTanka** [comment]
### Community Review — Add PR114 RunPod H100 SXM non-record submission

**BPB:** 1.1554 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `013954d4c19b`, file `records/track_non_record_16mb/2026-03-21_PR114_RunPod_H100SXM_TripleRerun/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.04s, dim=512, layers=9, vocab=1024, code=53572 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.04s, dim=512, layers=9, vocab=1024, code=53572 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
